atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -88,7 +91,25 @@ private static class State {
     public HikariCPConnectionManager(ConnectionConfig connConfig) {
         this.connConfig = Preconditions.checkNotNull(connConfig, "ConnectionConfig must not be null");
         this.hikariConfig = connConfig.getHikariConfig();
-        hikariConfig.setPoolName(hikariConfig.getPoolName() + "-" + uniquePoolId.incrementAndGet());
+        this.exec = makeExecutorIfNecessary();
+    }
+
+    /**
+     * Manage our own executor so that hikari does not manage it for us. This is
+     * a workaround for the executor leaking in hikari if there is a metric
+     * conflict.
+     */
+    private ScheduledExecutorService makeExecutorIfNecessary() {
```

Comment:
Does (some of) this need to be synchronized? Basically certain users of AtlasDB could end up hitting this in parallel if they create more than one TM.

Reply:
The executor is local to an instance of this class, and this method is only called by the constructor.

Reply:
fair enough - yeah makes sense